web_src/js/index.js

```diff
@@ -585,6 +585,35 @@ function initInstall() {
   });
 }
 
+function initIssueComments() {
+  if ($('.repository.view.issue .comments').length === 0) return;
+
+  $(document).click((event) => {
+    const urlTarget = $(':target');
+    if (urlTarget.length === 0) return;
+
+    const urlTargetId = urlTarget.attr('id');
+    if (!urlTargetId) return;
+    if (!/^(issue|pull)(comment)?-\d+$/.test(urlTargetId)) return;
+
+    const $target = $(event.target);
+
+    if ($target.closest(`#${urlTargetId}`).length === 0) {
+      if (!window.location.hash) return;
+
+      if (window.location.hash.length > 0 && window.location.hash !== '#') {
+        const i = window.location.toString().indexOf('#');
+
+        if (i >= 0) {
+          const scrollPosition = $(window).scrollTop();
+          window.location.hash = '';
```

Comment:
This leaves the empty `#` behind. Can you maybe try getting it to work using `replaceState` which does not suffer from this? I assume there must be a way to clear `:target` somehow.

Reply:
```js
window.history.replaceState(null, null, ' ');
```

this removes target url but does not affect `:target` effect

Reply:
You can just combine both methods and use `replaceState` to get rid of the final `#`:

````js
const scrollPosition = $(window).scrollTop();
window.location.hash = '';
$(window).scrollTop(scrollPosition);
window.history.replaceState(null, null, ' ');

Reply:
@silverwind  brilliant! that works as expected now. done. thank you :)